ignite/services/network/share_percent.go

```diff
@@ -0,0 +1,102 @@
+package network
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type SharePercents []SharePercent
+
+func (sp SharePercents) Empty() bool {
+	return len(sp) == 0
+}
+
+// SharePercent represent percent of total share
+type SharePercent struct {
+	denom string
+	// in order to avoid using numbers with floating point
+	// fractional representation is used: 297/10000 instead of 2.97%
+	nominator, denominator uint64
+}
+
+// Share returns coin share of total according to underlying percent
+func (p SharePercent) Share(total uint64) (sdk.Coin, error) {
+	resultNominator := total * p.nominator
+	if resultNominator%p.denominator != 0 {
+		err := fmt.Errorf("%s share from total %d is not integer: %f",
+			p.denom,
+			total,
+			float64(resultNominator)/float64(p.denominator),
+		)
+		return sdk.Coin{}, err
+	}
+	return sdk.NewInt64Coin(p.denom, int64(resultNominator/p.denominator)), nil
+}
+
+// SharePercentFromString parses share percent from string
+// format: 11.87%foo
+func SharePercentFromString(str string) (SharePercent, error) {
+	// validate raw percentage format
+	var rePercentageRequired = regexp.MustCompile(`^[0-9]+.[0-9]*%`)
```

Comment:
Can we move this line above  SharePercentFromString, so it wont get recompiled again and again when `services/network` used programmatically?